utils/schemed-kafka-components-utils/src/main/scala/pl/touk/nussknacker/engine/schemedkafka/source/UniversalKafkaSourceFactory.scala

```diff
@@ -256,7 +271,14 @@ class UniversalKafkaSourceFactory(
           .extractParameterForTests(parsedSchema)
           .map(_.toParameters)
           .map { params =>
-            KafkaTestParametersInfo(params, prepareTestRecord(runtimeSchema, universalSchemaSupport, topic))
+            val enrichedParams = params.collect {
+              case param if defaultValuesForTestParameters.contains(param.name) =>
+                param.copy(
+                  defaultValue = Some(defaultValuesForTestParameters(param.name))
```

Comment:
Maybe just `defaultValuesForTestParameters.get(param.name)` ? It will return Option. Then we wouldn't need collect and pattern matching, just a simple map.

Reply:
I changed `collect` to `map`. Some parameters can already have a default value determined, so I'm not calling the copy method for all parameters to not set a None value for them.